Write a routine for defining internals/private params

When the params are defined in a traditional way, shellcheck checks if the params are used in the current `.sh` file. Very often they are not, so in these cases it's necessary to use the shellcheck directive `shellcheck disable=SC2034`. To make defining internals/private params more neat, write a special routine and name it `def_priv_var`.

--- a/essentials.sh
+++ b/essentials.sh
@@ -185,6 +185,21 @@ def_protected_var() {
     >&2 echo "+ ${var_name}=*****"
 }
 
+# Defines a private variable. The function helps avoid using shellcheck disable=SC2034
+# Globals:
+#     None
+# Arguments:
+#     Variable name
+#     Value by default
+# Returns:
+#     None
+def_private_var() {
+    local var=$1
+    local val=$2
+
+    eval ${var}="\"${val}\""
+}
+
 # Acts like def_var but additionally requires that the target value is Boolean.
 # Globals:
 #     None

--- a/pieman.sh
+++ b/pieman.sh
@@ -143,48 +143,39 @@ def_bool_var XFCE4 false
 # Internal params
 #
 
-R=${BUILD_DIR}/${PROJECT_NAME}/chroot
+def_private_var R "${BUILD_DIR}/${PROJECT_NAME}/chroot"
 
-BOOT=${BUILD_DIR}/${PROJECT_NAME}/boot
+def_private_var BOOT "${BUILD_DIR}/${PROJECT_NAME}/boot"
 
-MOUNT_POINT=${BUILD_DIR}/${PROJECT_NAME}/mount_point
+def_private_var MOUNT_POINT "${BUILD_DIR}/${PROJECT_NAME}/mount_point"
 
-# shellcheck disable=SC2034
-ETC=${R}/etc
+def_private_var ETC "${R}/etc"
 
-# shellcheck disable=SC2034
-USR_BIN=${R}/usr/bin
+def_private_var USR_BIN "${R}/usr/bin"
 
-ARTIFACT="${PIEMAN_DIR}/${BUILD_DIR}/${PROJECT_NAME}.mender"
+def_private_var ARTIFACT "${PIEMAN_DIR}/${BUILD_DIR}/${PROJECT_NAME}.mender"
 
-# shellcheck disable=SC2034
-BASE_PACKAGES=""
+def_private_var BASE_PACKAGES ""
 
-BUILD_TYPE="${IMAGE_CLASSIC}"
+def_private_var BUILD_TYPE "${IMAGE_CLASSIC}"
 
-# shellcheck disable=SC2034
-FIRSTBOOT="/tmp/firstboot-${PROJECT_NAME}.sh"
+def_private_var FIRSTBOOT "/tmp/firstboot-${PROJECT_NAME}.sh"
 
-IMAGE=${BUILD_DIR}/${PROJECT_NAME}.img
+def_private_var IMAGE "${BUILD_DIR}/${PROJECT_NAME}.img"
 
-# shellcheck disable=SC2034
-KEYRING="/tmp/atomatically-generated-keyring-for-${PROJECT_NAME}.gpg"
+def_private_var KEYRING "/tmp/atomatically-generated-keyring-for-${PROJECT_NAME}.gpg"
 
-# shellcheck disable=SC2034
-PM_OPTIONS=""
+def_private_var PM_OPTIONS ""
 
-# shellcheck disable=SC2034
-EXIT_REQUEST="EXIT"
+def_private_var EXIT_REQUEST "EXIT"
 
-# shellcheck disable=SC2034
-REDIS_IS_AVAILABLE=true
+def_private_var REDIS_IS_AVAILABLE true
 
-TOOLSET_FULL_PATH="${TOOLSET_DIR}/${TOOLSET_CODENAME}"
+def_private_var TOOLSET_FULL_PATH "${TOOLSET_DIR}/${TOOLSET_CODENAME}"
 
-SOURCE_DIR=devices/${DEVICE}/${OS}
+def_private_var SOURCE_DIR "devices/${DEVICE}/${OS}"
 
-# shellcheck disable=SC2034
-YML_FILE=${SOURCE_DIR}/pieman.yml
+def_private_var YML_FILE "${SOURCE_DIR}/pieman.yml"
 
 run_scripts "helpers"
 

--- a/test/test_essentials.sh
+++ b/test/test_essentials.sh
@@ -26,6 +26,19 @@ setUp() {
 # Let the test begin
 #
 
+test_defining_private_variables() {
+    priv="test"
+
+    def_private_var priv "default"
+    assertEquals "default" "${priv}"
+
+    def_private_var priv2 "default"
+    assertEquals "default" "${priv2}"
+
+    def_private_var EMPTY ""
+    assertNull "${EMPTY}"
+}
+
 test_defining_variables() {
     S1="test"
     def_var S1 "value by default"

